charms/worker/k8s/src/pki.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""A module providing PKI related functionalities."""
+
+import ipaddress
+import logging
+import os
+import typing
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509.extensions import ExtensionNotFound
```

Comment:
sigh... we can't use this :( as it depends on libraries that are series specific.  

grow to love `subprocess.check_output()`